src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,22 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        UserData userData = new UserData(userId, "Details for user" + userId);
+        cache.put(userId, userData);
+        return CompletableFuture.supplyAsync(()
+                -> userData);
```

Comment:
The 'executor' field is not being used in the 'CompletableFuture.supplyAsync' method. If the intention was to run the 'supplyAsync' on the provided executor, you should pass 'executor' as the second parameter to the 'supplyAsync' method.